tests/test_io.py

```diff
@@ -18,42 +18,89 @@
 """IO functionality tests"""
 
 import json
+from pathlib import Path
 
 import pytest
+import yaml
 
 from ghga_transpiler import io
 
-from .fixtures.test_data_objects.conversion_data import EXPECTED_CONVERSION
+from .fixtures.test_data_objects.conversion_data import (
+    EXPECTED_CONVERSION_DATAPACK,
+    EXPECTED_CONVERSION_JSON,
+)
 
 
-def test_write_json_file(tmp_path):
-    """Test write_json"""
+def test_write_datapack_json(tmp_path: Path):
+    """Test write_datapack in json format"""
     out_path = tmp_path.joinpath("out.json")
-    io.write_json(data=EXPECTED_CONVERSION, path=out_path, force=False)
+    io.write_datapack(
+        data=EXPECTED_CONVERSION_DATAPACK, path=out_path, yaml_format=False, force=False
+    )
 
     with open(file=out_path, encoding="utf8") as in_file:
         data = json.load(fp=in_file)
-    assert data == EXPECTED_CONVERSION
+    assert data == EXPECTED_CONVERSION_JSON
 
 
-def test_write_json_file_force(tmp_path):
+def test_write_datapack_yaml(tmp_path: Path):
+    """Test write_datapack in json format"""
+    out_path = tmp_path.joinpath("out.json")
+    io.write_datapack(
+        data=EXPECTED_CONVERSION_DATAPACK, path=out_path, yaml_format=True, force=False
+    )
+
+    with open(file=out_path, encoding="utf8") as in_file:
+        data = yaml.safe_load(in_file)
+    assert data == EXPECTED_CONVERSION_JSON
+
+
+def test_write_datapack_json_force(tmp_path: Path):
     """Test write_json overwrite of output"""
     out_path = tmp_path.joinpath("out.json")
     out_path.touch()
-    io.write_json(data=EXPECTED_CONVERSION, path=out_path, force=True)
+    io.write_datapack(
+        data=EXPECTED_CONVERSION_DATAPACK, path=out_path, yaml_format=False, force=True
+    )
 
 
-def test_write_json_file_no_force(tmp_path):
-    """Test write_json abort if output exists"""
+def test_write_datapack_yaml_force(tmp_path):
+    """Test write_json overwrite of output"""
     out_path = tmp_path.joinpath("out.json")
     out_path.touch()
-    with pytest.raises(FileExistsError):
-        io.write_json(data=EXPECTED_CONVERSION, path=out_path, force=False)
+    io.write_datapack(
+        data=EXPECTED_CONVERSION_DATAPACK, path=out_path, yaml_format=True, force=True
+    )
 
 
-def test_write_json_file_stdout(capfd):
-    """Test write_json overwrite of output"""
-    io.write_json(data=EXPECTED_CONVERSION, path=None, force=True)
+def test_write_datapack_json_no_force(tmp_path: Path):
+    """Test write_datapack abort if output exists"""
+    out_path = tmp_path.joinpath("out.yaml")
```

Comment:
```suggestion
    out_path = tmp_path.joinpath("out.json")
```

Should match `yaml_format=False`.